kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/inband/ParcelTest.java

```diff
@@ -65,4 +81,141 @@ void shouldRoundTrip(Set<RecordField> fields, Record record) {
         assertThat(buffer.remaining()).isEqualTo(0);
     }
 
+    private record Header(@JsonProperty(required = true) String keyBase64, String valueBase64) {
```

Comment:
AFAIK Jackson datamapper will deserial from Base64 to a byte[] field, so you should be able to remove your methods.